paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/commands/WriteIntoPaimonTable.scala

```diff
@@ -83,10 +84,19 @@ case class WriteIntoPaimonTable(
     val commitMessages = writer.write(data)
     writer.commit(commitMessages)
 
-    markDoneIfNeeded(commitMessages)
+    preFinish(commitMessages)
     Seq.empty
   }
 
+  private def preFinish(commitMessages: Seq[CommitMessage]): Unit = {
+    val coreOptions = table.coreOptions();
+    if (coreOptions.tagCreationMode() == TagCreationMode.BATCH) {
+      val tagCreation = new TagBatchCreation(table)
+      tagCreation.createTag()
+    }
+    markDoneIfNeeded(commitMessages)
+  }
+
   private def markDoneIfNeeded(commitMessages: Seq[CommitMessage]): Unit = {
     val coreOptions = table.coreOptions()
     if (coreOptions.toConfiguration.get(CoreOptions.PARTITION_MARK_DONE_WHEN_END_INPUT)) {
```

Comment:
ditto

Reply:
this option is used several times, I think it's ok here

Reply:
My mistake.